src/helper/getIntersectedLinesAndPoint.js

```diff
@@ -15,14 +15,25 @@ const getIntersectedLinesAndPoint = (coordinate, lines, map, snapTolerance) => {
   const isPointAlreadyExist = {};
   const mousePx = map.getPixelFromCoordinate(coordinate);
 
-  lines.forEach((lineA) => {
-    lines.forEach((lineB) => {
-      const intersections = OverlayOp.intersection(
-        parser.read(lineA.getGeometry()),
-        parser.read(lineB.getGeometry()),
-      );
+  const parsedLines = lines.map((line) => [
+    line,
+    parser.read(line.getGeometry()),
+  ]);
+  parsedLines.forEach(([lineA, parsedLineA]) => {
+    parsedLines.forEach(([lineB, parsedLineB]) => {
+      if (lineA === lineB || isSameLines(lineA, lineB, map)) {
+        return;
+      }
+
+      let intersections;
+      try {
+        intersections = OverlayOp.intersection(parsedLineA, parsedLineB);
+      } catch (e) {
+        return; // The OverlayOp will sometimes error with topology errors for certain lines
```

Comment:
@megawac  Have you some more precise info about those topology error ? Could be nice to have an example.

Reply:
I seem to recall them occurring quite often when #267 was occurring... I'll see if I can track down if there was more context on internal tickets.

Reply:
![image](https://github.com/geops/openlayers-editor/assets/3475472/36526b22-2817-48b1-bcd8-4a826f3c8193)

This is from the original ticket reported on our side. So in the case the case of the error, `parsedLineB` snap line appears to be invalid from the error below.

```
Uncaught TopologyException: found non-noded intersection between LINESTRING ( 9.952973188580247 2.35845900632583, 11.739231890263671 2.3680559374052663 ) and LINESTRING ( 9.952973188580248 2.35845900632583, 11.452360499376073 508.73944164968873 ) [ (9.952973188580248, 2.35845900632583, NaN) ]
    at g.checkValid (bundle.js)
    at E.checkValid (bundle.js)
    at E.checkValid (bundle.js)
    at A.computeOverlay (bundle.js)
    at A.getResultGeometry (bundle.js)
    at A.overlayOp (bundle.js)
    at E.getResultGeometry (bundle.js)
    at E.overlayOp (bundle.js)
    at A.intersection (bundle.js)
    at bundle.js
    at Array.forEach (<anonymous>)
    at bundle.js
    at Array.forEach (<anonymous>)
    at J (bundle.js)
    at ee.drawSnapLines (bundle.js)
    at ee.onMove (bundle.js)
    at a.handleEvent (bundle.js)
    at le.handleMapBrowserEvent (bundle.js)
    at u.dispatchEvent (bundle.js)
    at u.relayMoveEvent_ (bundle.js)
    at HTMLDivElement.d (bundle.js)
```